dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -179,23 +179,87 @@ def _add_helper_merge_profile_histograms(
         # only the methods which intersect should exist.
         self.histogram_bin_method_names = bin_methods
         self.histogram_methods = dict()
+
+        # Set ideal bin count w/ if statement of some sort
+        ideal_count_of_bins = self.min_histogram_bin
+        if self.user_set_histogram_bin is not None:
+            ideal_count_of_bins = self.user_set_histogram_bin
+
         for method in self.histogram_bin_method_names:
             self.histogram_methods[method] = {
                 "total_loss": 0,
                 "current_loss": 0,
                 "histogram": {"bin_counts": None, "bin_edges": None},
             }
 
-        combined_values = np.concatenate(
-            [other1._histogram_to_array(), other2._histogram_to_array()]
+        # calculate the min of the first edge and the max of the last edge
+        # between two arrays
+        if not (other1.min is None or other2.min is None):
+            global_min_of_histogram_edges = min(other1.min, other2.min)
```

Comment:
bc this is an or, what happens if one of them is `None`. does this error?/

Reply:
if one them is None the whole expression is false so no it does not. If the expression was 
`not (other1.min is None and other2.min is None)` we may have an issue

Reply:
at this point we have the `self.min` I think since we moved the calc below.

We could do:
```python
global_min_of_histogram_edges = self.min
if self.min is None:
    global_min_of_histogram_edges = min(
        other1._stored_histogram["histogram"]["bin_edges"][0],
        other2._stored_histogram["histogram"]["bin_edges"][0],
    )
```
it removes the `else` and utilizes existing functionality.

This is conditional on `self.min` being updated prior

Reply:
else is needed. mypy fails because of  requirements from lispace parameter types

Reply:
We shouldn't be changing code to meet mypy expectations. I'd defer to cleanliness and readability over mypy, but it typically is a good guide towards that end.

Reply:
It failing from that change also would indicate an issue in our description / typing else where.

Reply:
I figured it out, it was bc i wrote `if self.min is None` needs to be `if global_min_of_histogram_edges is None`

```python
global_min_of_histogram_edges = self.min
        if global_min_of_histogram_edges is None:
            global_min_of_histogram_edges =  min(
                other1._stored_histogram["histogram"]["bin_edges"][0],
                other2._stored_histogram["histogram"]["bin_edges"][0],
            )
```